app/controllers/candidate_interface/referees_controller.rb

```diff
@@ -12,15 +12,31 @@ def index
     end
 
     def type
-      @reference_type_form = Reference::RefereeTypeForm.new
+      if params[:id]
+        set_referee_id
+
+        @reference_type_form = Reference::RefereeTypeForm.build_from_reference(@referee)
+      else
+        @reference_type_form = Reference::RefereeTypeForm.new
+      end
     end
 
     def update_type
       @reference_type_form = Reference::RefereeTypeForm.new(referee_type: referee_type_param)
 
-      return render :type unless @reference_type_form.valid?
+      if params[:id]
+        set_referee_id
+
+        return redirect_to action: 'type', id: @id unless @reference_type_form.valid?
+
+        @reference_type_form.save(@referee)
+
+        redirect_to candidate_interface_review_referees_path
```

Comment:
Personally I find it easier to follow a simple `if` statement than a `return ... unless ...`. What do you think of the below instead?

```suggestion
        set_referee_id

        if @reference_type_form.valid?
          @reference_type_form.save(@referee)

          redirect_to candidate_interface_review_referees_path
        else
          render :type
        end
```

Reply:
I tried to avoid nesting more if statements within `if params[:id]`, we need to render `:type` with param if there is an `id`.